storage-inventory/ops/README.md

```diff
@@ -2,118 +2,169 @@
 
 # CADC Storage Inventory Architecture and Deployment
 
-
-Read the entire document or jump to a relevant section below.
-
-
-### Table of Contents
-
-1. [Introduction](#introduction)
-    * [Key Storage Inventory Terms and Concepts](#key-storage-inventory-terms-and-concepts)
-    * [Storage Inventory Resources](#storage-inventory-resources)
-1. [Overview of a Storage Site](#overview-of-a-storage-site)
-    * [System Architecture](#system-architecture)
-    * [Standalone Storage Site](#standalone-storage-site)
-1. [Client Software](#client-software)
-1. [Deployment Prerequisites](#deployment-prerequisites)
-    * [Hardware Requirements](#hardware-requirements)
-    * [Software Requirements](#software-requirements)
-1. [Deployment](#deployment)
-1. [Healthchecks and Monitoring](#healthchecks-and-monitoring)    
-1. [FAQ](#faq)
-
-
 ## Introduction
+### Storage Inventory Terms and Concepts
+<a id='term-artifact'>**artifact**</a>  
+A representation of a file and its metadata in the SI database, specifically the _inventory.Artifact_ table (see the [SI data model](https://github.com/opencadc/storage-inventory/tree/master/storage-inventory-dm)).  The term _artifact_ is often used to refer to both the database representation of a file and the file itself as one thing, but when an SI application is acting on an _artifact_, it specifically refers to the database.
+- ➡️ an artifact _URI_ is a unique idenitfier for a file (or object) stored in SI.  The last path component of a _URI_ is always the _filename_.
+
+<a id='term-bucket'>**bucket**</a>  
+SI can organize units of work (number of files or artifacts) by `buckets` and applications can be configured to work on subsets of these buckets.  
+- Buckets are represented by hex strings, e.g. `a74`.
+- When applications refer to buckets on storage, it refers to the location (e.g., an S3 bucket or a directory) in which files are stored.  This is dependent upon your storage configuration.
+- When applications refer to buckets in the SI database, it refers to the _inventory.Artifact.uriBucket_ column in the database.  This column is populated with a random hex string on artifact creation.
+- ❗ _storage_ buckets and _uri_ buckets are not the same thing, e.g. files in _storage_ bucket `04c` are not necessarily the same as artifacts in _uri_ bucket `04c`.
```

Comment:
The term 'bucket' is used in storage to mean two different things: an abstraction of storage location and a artifact label. Both are hex numbers.  So, a file could have two different 'buckets' associated with it: a storage bucket (which might or might not be the same S3 or Swift bucket) and an artifact bucket, and those two aren't necessarily the same hex number -- in fact, it would be very unlikely for them to be the same.

The term bucket is used for S3 and Swift storage, independently of SI (e.g., from the man page of `s3cmd`: _s3cmd ls [s3://BUCKET[/PREFIX]] --  List objects or buckets_).  This might be a bit confusing when we're also talking about SI buckets.  I'll try to clear up the text a bit, but we do need to clarify the difference between storage buckets and artifact buckets (and, as you point out, S3/Swift buckets).

Reply:
Can we just highlight that the SI buckets are simply units of work in SI and have nothing to do with the buckets that might be defined in the underlying storage system (S3 and swift)?

Reply:
I've changed the language to emphasize that the SI buckets are just labels.  The text describing the relationship between SI buckets and storage location was in answer to a question that came up in an SRC discussion, but so I've kept that but again pushed the label aspect.  The difference between an SI storage bucket and SI artifact bucket is also something that caused confusion elsewhere, so I've kept that text.

Reply:
The `Artifact.uriBucket` is always a hex-string (currently hard coded to 5 chars so the set of artifacts can be subdivided into max 1e6 buckets, or fewer/larger buckets).

The `StorageLocation.storageBucket` is an implementation detail. See BucketType in cadc-storage-adapter interface. 

The OpaqueFileSystemStorageAdapter and the SwiftStorageAdapter they are HEX (length configurable) but that doesn't necessarily mean a specific number of storage objects of a certain type. The AdStorageAdapter has bucket type PLAIN. If we ever finish the "not opaque" FileSystemStorageAdapter, it will be PATH (or something like that).

Reply:
Practically, `Artifact.uriBucket` is the thing being used when `critwall` and  `ratik` are configured to work on buckets. It is feasible to make `fenwick` and use this, but I don't think it is needed to scale incremental sync.

Only `tantar` uses about `StorageLocation.storageBucket`.

Although we used the configuration term _bucket_ for all of these uses, I'm not sure that was wrong... it is the same concept and in the context of the above applications it is about parallelizing work loads for scalability and nothing more.

It is only in the context of a StorageAdapter configuration that "buckets" may map to something in the back end storage and deployers need to be aware of the pros and cons.

Reply:
aside: I would not mention S3 because we gave up on it due to insufficient API

Reply:
Fair enough: I only included it because the concept of S3 buckets is more widely known, but It could confuse someone.  Removed reference to S3.

Reply:
The confusion comes from overloading the word _bucket_. The apps config files refer to `buckets` only so one have to figure out which ones is that. I didn't realize this from the beginning but I wish we used a different term for artifact URI grouping (`uriBin` or something similar) but it's probably too late to do that now.
@zombiepig01 - in this case just make clear that there are 2 types of independent buckets in the system: `uriBucket` (units of work) and `storageBuckets` (units of storage) and describe the role of each of them.

Reply:
I think we're getting into details that are best explained in the dev documents for the adapters or model.  Remember, this are operator/installer documents, not developer documents.

A bucket in the SI config is an abstraction, a label: 
- it is used as a unit of work (for applications working on storage and those working on uris).
- it is a label which in the storage model.  On some storage platforms, this label might be used in the storage organization, too (Swift, Opaque fs adapter).  
- a storage bucket is not the same as a uri bucket

That is basically what I say in the docs, plus a few other points of clarification that are to address questions that came up in other discussions.   There are a couple other storage adapters that are either not going to be used by anyone else (AD) or are not yet fully developed (structured FS adapter) so I'm not going to worry about those yet.

Reply:
FWIW, critwall and tantar refer to buckets: these are storage buckets as defined in the storage-adapter properties fileless ca.
- org.opencadc.critwall.buckets
- org.opencadc.tantar.buckets

ratik is the only other application that refers to buckets, and it is refering to uri buckets:
- org.opencadc.ratik.buckets